lib/data/locationConstraintParser.js

```diff
@@ -64,14 +65,16 @@ function parseLC() {
                 s3Params.secretAccessKey =
                     locationObj.details.credentials.secretKey;
             }
-            clients[location] = new AwsClient({
+            const config = {
                 s3Params,
                 awsBucketName: locationObj.details.bucketName,
```

Comment:
I'm also generally a bit confused about how the config will look for a GCP location constraint. With this setup, the user will have to have an AWS account name and access key as well as GCP credentials, correct?

Also, with the two location constraint types sharing these config options, maybe we should rename awsBucketNmae to just bucketName?

Reply:
Discussed credentials, looks like only GCP creds are needed. Still think the `awsBucketName` key should be changed :)

Reply:
Yup. For the config, we will just need the GCP interoperability credentials. The credentials are very similar to that of AWS's, an access id and a secret key, and with it, the location constraint for GCP will just be:
```
"gcpbackend": {
        "type": "gcp",
        "legacyAwsBehavior": true,
        "details": {
            "awsEndpoint": "storage.googleapis.com",
            "bucketName": "bucket name",
            "bucketMatch": true,
            "credentialsProfile": "profile name"
        }
    }
```  
Those credentials will be store as one of the profiles in AWS, so retrieving it will just be specifying the profile name.  
As for the variable names, yup... I had overlooked that